Dockerfile

```
@@ -1,41 +1,60 @@
-# NOTE: Testing Only. DO NOT use this in production
-
-ARG NODE_VERSION=16-alpine
-
-FROM node:${NODE_VERSION}
-
-# Required to run OracleDB
-# Technically not required for the others, but I'd rather have 1 image that works for all, instead of building n images
-# per test
-#WORKDIR /tmp
-#RUN apk --no-cache add libaio libnsl libc6-compat curl && \
-#    curl -o instantclient-basiclite.zip https://download.oracle.com/otn_software/linux/instantclient/instantclient-basiclite-linuxx64.zip -SL && \
-#    unzip instantclient-basiclite.zip && \
-#    mv instantclient*/ /usr/lib/instantclient && \
-#    rm instantclient-basiclite.zip && \
-#    ln -s /usr/lib/instantclient/libclntsh.so.19.1 /usr/lib/libclntsh.so && \
-#    ln -s /usr/lib/instantclient/libocci.so.19.1 /usr/lib/libocci.so && \
-#    ln -s /usr/lib/instantclient/libociicus.so /usr/lib/libociicus.so && \
-#    ln -s /usr/lib/instantclient/libnnz19.so /usr/lib/libnnz19.so && \
-#    ln -s /usr/lib/libnsl.so.2 /usr/lib/libnsl.so.1 && \
-#    ln -s /lib/libc.so.6 /usr/lib/libresolv.so.2 && \
-#    ln -s /lib64/ld-linux-x86-64.so.2 /usr/lib/ld-linux-x86-64.so.2
-#
-#ENV ORACLE_BASE /usr/lib/instantclient
-#ENV LD_LIBRARY_PATH /usr/lib/instantclient
-#ENV TNS_ADMIN /usr/lib/instantclient
-#ENV ORACLE_HOME /usr/lib/instantclient
+# syntax=docker/dockerfile:1.4
 
-WORKDIR /directus
+####################################################################################################
+## Prepare Workspace
+
+FROM node:18-alpine AS workspace
+WORKDIR /workspace
+
+COPY pnpm-lock.yaml .
+
+RUN corepack enable \
+	&& corepack prepare pnpm@7.30.0 --activate \
+	&& pnpm fetch
 
 COPY . .
 
-RUN apk add --update python3 make g++\
-   && rm -rf /var/cache/apk/*
+RUN pnpm install --recursive --offline --frozen-lockfile
+
+####################################################################################################
+## Create Deployment
+
+FROM workspace AS pruned
+WORKDIR /workspace
+ENV NODE_OPTIONS=--max-old-space-size=8192
+
+RUN pnpm --recursive run build \
+	&& pnpm --filter directus deploy --prod pruned
+
+####################################################################################################
+## Create Production Image
+
+FROM node:18-alpine
+
+RUN mkdir /directus \
+	&& echo "update-notifier=false" >> ~/.npmrc \
+	&& mkdir -p /directus/data/database /directus/data/extensions /directus/data/uploads \
+	&& chown node:node /directus/data;
+
+WORKDIR /directus
+
+EXPOSE 8055
+
+ENV DB_CLIENT="sqlite3"
+ENV DB_FILENAME="/directus/data/database/database.sqlite"
+ENV EXTENSIONS_PATH="/directus/data/extensions"
+ENV STORAGE_LOCAL_ROOT="/directus/data/uploads"
+ENV NODE_ENV="production"
```

I think setting the `update-notifier` option via the file has no effect (anymore) since it's placed at `/root/.npmrc` but should be placed at `/home/node/.npmrc`...

However a nicer variant IMO would be to set the option via an environment variable anyway which should work according to https://stackoverflow.com/a/72095210/20504834:

```suggestion
	&& mkdir -p /directus/data/database /directus/data/extensions /directus/data/uploads \
	&& chown node:node /directus/data;

WORKDIR /directus

EXPOSE 8055

ENV DB_CLIENT="sqlite3"
ENV DB_FILENAME="/directus/data/database/database.sqlite"
ENV EXTENSIONS_PATH="/directus/data/extensions"
ENV STORAGE_LOCAL_ROOT="/directus/data/uploads"
ENV NODE_ENV="production"
ENV NPM_CONFIG_UPDATE_NOTIFIER="false"
```